week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product/Product.ts

```diff
@@ -0,0 +1,96 @@
+import {v4} from 'uuid';
+import {IProduct, IProductWithPrice, PRODUCT_TYPE} from './Product.types';
+
+class Product {
+  private readonly _id: string;
+  private _name: string;
+  private _amount: number;
+
+  public constructor({ name, amount }: IProduct) {
+    this._id = v4();
+    this._name = name;
+    this._amount = amount;
```

Comment:
Jeśli Twój zamysł był taki żeby nie dało się wrzucić ujemnej wartości `amount` to powinno tutaj być:

```javascript
this.amount = amount
```
Tak aby uruchomiła się logika `settera` dla `amount`

Bo teraz nikt mi nie zabroni zrobić tak:
![image](https://user-images.githubusercontent.com/25398949/193880127-4c25eb1b-136e-412e-9285-3d0498da06de.png)

i to zadziała :)

Reply:
Faktycznie, wpadłem w pułapkę tego zapisu z `_`